src/components/Pane/Pane.tsx

```diff
@@ -0,0 +1,224 @@
+/* tslint:disable:no-unused-variable */
+import * as React from 'react';
+/* tslint:enable:no-unused-variable */
+
+import { BaseComponent } from '../../common/BaseComponent';
+import { IPaneProps, PaneType } from './Pane.Props';
+import { Popup } from '../Popup/index';
+import { css } from '../../utilities/css';
+import { getId } from '../../utilities/object';
+import { getRTL } from '../../utilities/rtl';
+import { PaneContent } from './PaneContent';
+import { WrappedContent } from './WrappedContent';
+import './Pane.scss';
+
+export interface IPaneState {
+  isOpen?: boolean;
+  isAnimatingOpen?: boolean;
+  isAnimatingClose?: boolean;
+  id?: string;
+}
+
+export class Pane extends BaseComponent<IPaneProps, IPaneState> {
+
+  public static defaultProps: IPaneProps = {
+    isOpen: false,
+    hasCloseButton: true,
+    type: PaneType.small
+  };
+
+  private _contentContainer: HTMLElement;
+  private _mainContent: HTMLElement;
+  private _initialContentWidth: number;
+
+  constructor(props: IPaneProps) {
+    super(props);
+
+    this._onClose = this._onClose.bind(this);
+    this._onPaneRef = this._onPaneRef.bind(this);
+
+    this.state = {
+      isOpen: !!props.isOpen,
+      isAnimatingOpen: props.isOpen,
+      isAnimatingClose: false,
+      id: getId('Pane')
+    };
+  }
+
+  public componentDidMount() {
+    if (this.props.isOverlay) {
+      // Set original content width for overlay mode
+      this._initialContentWidth = this._contentContainer.clientWidth;
+    }
+
+    if (this.state.isOpen) {
+      this._async.setTimeout(() => {
+        this.setState({
+          isAnimatingOpen: false
+        });
+      }, 2000);
+    }
+  }
+
+  public componentWillReceiveProps(newProps: IPaneProps) {
+    if (newProps.isOpen !== this.state.isOpen) {
+      this.setState({
+        isOpen: newProps.isOpen,
+        isAnimatingOpen: newProps.isOpen ? true : false,
+        isAnimatingClose: newProps.isOpen ? false : true
+      });
+    }
+  }
+
+  public componentDidUpdate() {
+    if (this.props.isOverlay) {
+      // Use original content width for overlay mode
+      this._contentContainer.style.width = this._initialContentWidth + 'px';
+    } else {
+      // Viewport content width for push mode
+      this._contentContainer.style.width = this._getContainerWidth() + 'px';
+    }
+  }
+
+  public render() {
+    let { className = '', type, hasCloseButton, headerText, closeButtonAriaLabel, headerClassName = ''  } = this.props;
+    let { isOpen, isAnimatingOpen, isAnimatingClose, id } = this.state;
+    let isRTL = getRTL();
+    let isOnRightSide = !isRTL;
+    const headerTextId = id + '-headerText';
+    let pendingCommandBarContent = '';
+
+    let header;
```

Comment:
I was just copying a style I had seen in Panel.  I switch it back to my preference.